staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/defaults/defaults.go

```diff
@@ -95,7 +95,7 @@ func (b BuildDefaults) applyPodProxyDefaults(pod *corev1.Pod, isCustomBuild bool
 	for _, c := range allContainers {
 		// All env vars are allowed to be set in a custom build pod, the user already has
 		// total control over the env+logic in a custom build pod anyway.
-		externalEnv := make([]corev1.EnvVar, 3)
+		externalEnv := []corev1.EnvVar{}
```

Comment:
@adambkaplan this is the fix for the custom build extended test error

Reply:
Good catch - iirc you can get interesting behavior around `append` with slices.